packages/elements/src/tree/helpers/renderer.ts

```diff
@@ -11,23 +12,23 @@ type RendererScope = {
 };
 
 export class TreeRenderer extends Renderer {
-  constructor (scope?: unknown) {
 
-    let manager: TreeManager<TreeDataItem>;
-    let currentMode: TreeManagerMode;
-    let currentComposer: CollectionComposer<TreeDataItem>;
+  /**
+   * Renderer key prefix, used in combination with item value to give unique id to each item
+   */
+  public key = uuid().split('-')[0];
+
+  constructor (scope?: unknown) {
     super((item: TreeDataItem, composer: CollectionComposer<TreeDataItem>, element = document.createElement('ef-tree-item')): HTMLElement => {
       const el = element as TreeItem;
       const multiple = !!scope && (scope as RendererScope).multiple === true;
       const noRelation = !!scope && (scope as RendererScope).noRelation === true;
       const mode = !multiple || !noRelation ? TreeManagerMode.RELATIONAL : TreeManagerMode.INDEPENDENT;
-      if (currentComposer !== composer || currentMode !== mode) {
-        currentMode = mode;
-        currentComposer = composer;
-        manager = new TreeManager(composer, mode);
-      }
```

Comment:
Hi @TremayneChrist , I wonder what `currentMode` and `currentMode` are for and is it safe to remove. I want to add key property to `Renderer` but `super()` must be the first statement in constructor.

Reply:
This mode is for setting relational updates on/off. So if you toggle a parent it will decide whether to check/uncheck dependents.

Reply:
This also probably affects the selected state of items in the renderer, which is why it is required.

Reply:
But both `currentMode` and `currentComposer` are not used at all in the renderer, I've done some basic testing and see no difference.

Reply:
The mode switches the logic in the tree manager

Reply:
This code creates a new tree manager is the data/mode has changed